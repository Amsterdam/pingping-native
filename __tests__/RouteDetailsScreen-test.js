import React from 'react';

import { MockedProvider } from '@apollo/client/testing';
import { NativeBaseProvider } from 'native-base';
import renderer from 'react-test-renderer';

import RouteDetailScreen from '../src/screens/RouteDetailsScreen';

jest.useFakeTimers();
const navigation = { navigate: jest.fn() };

test('renders correctly', () => {
	const tree = renderer
		.create(
			<NativeBaseProvider>
				<MockedProvider mocks={[]}>
					<RouteDetailScreen
						navigation={navigation}
						route={{ params: { routeId: 1 } }}
					/>
				</MockedProvider>
			</NativeBaseProvider>,
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
