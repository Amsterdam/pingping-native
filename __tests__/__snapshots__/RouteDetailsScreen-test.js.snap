// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 400.2,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#E1E9EE",
            "overflow": "hidden",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": -350,
              },
            ],
          }
        }
      >
        <BVLinearGradient
          colors={
            Array [
              4292995566,
              4294113532,
              4292995566,
            ]
          }
          endPoint={
            Object {
              "x": 1,
              "y": 0,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "height": 10,
              "width": "30%",
            },
            Object {
              "backgroundColor": "#E1E9EE",
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -350,
                },
              ],
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4292995566,
                4294113532,
                4292995566,
              ]
            }
            endPoint={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "height": 40,
              "marginVertical": 15,
              "width": "80%",
            },
            Object {
              "backgroundColor": "#E1E9EE",
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -350,
                },
              ],
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4292995566,
                4294113532,
                4292995566,
              ]
            }
            endPoint={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "height": 15,
              "marginVertical": 2,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#E1E9EE",
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -350,
                },
              ],
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4292995566,
                4294113532,
                4292995566,
              ]
            }
            endPoint={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "height": 15,
              "marginVertical": 2,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#E1E9EE",
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -350,
                },
              ],
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4292995566,
                4294113532,
                4292995566,
              ]
            }
            endPoint={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "height": 15,
              "marginVertical": 2,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#E1E9EE",
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -350,
                },
              ],
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4292995566,
                4294113532,
                4292995566,
              ]
            }
            endPoint={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "height": 15,
              "marginVertical": 2,
              "width": "50%",
            },
            Object {
              "backgroundColor": "#E1E9EE",
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -350,
                },
              ],
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4292995566,
                4294113532,
                4292995566,
              ]
            }
            endPoint={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 25,
        }
      }
    >
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 4,
                "height": 15,
                "width": 120,
              },
              Object {
                "backgroundColor": "#E1E9EE",
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": -350,
                  },
                ],
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4292995566,
                  4294113532,
                  4292995566,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 4,
                "height": 15,
                "width": 120,
              },
              Object {
                "backgroundColor": "#E1E9EE",
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": -350,
                  },
                ],
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4292995566,
                  4294113532,
                  4292995566,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "height": 25,
              "marginTop": 25,
              "width": "40%",
            },
            Object {
              "backgroundColor": "#E1E9EE",
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -350,
                },
              ],
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4292995566,
                4294113532,
                4292995566,
              ]
            }
            endPoint={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 25,
        }
      }
    >
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 50,
                "height": 30,
                "marginRight": 20,
                "width": 30,
              },
              Object {
                "backgroundColor": "#E1E9EE",
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": -350,
                  },
                ],
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4292995566,
                  4294113532,
                  4292995566,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 20,
                "width": 650,
              },
              Object {
                "backgroundColor": "#E1E9EE",
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": -350,
                  },
                ],
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4292995566,
                  4294113532,
                  4292995566,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 25,
        }
      }
    >
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 50,
                "height": 30,
                "marginRight": 20,
                "width": 30,
              },
              Object {
                "backgroundColor": "#E1E9EE",
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": -350,
                  },
                ],
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4292995566,
                  4294113532,
                  4292995566,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 20,
                "width": 650,
              },
              Object {
                "backgroundColor": "#E1E9EE",
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": -350,
                  },
                ],
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4292995566,
                  4294113532,
                  4292995566,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 25,
        }
      }
    >
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 50,
                "height": 30,
                "marginRight": 20,
                "width": 30,
              },
              Object {
                "backgroundColor": "#E1E9EE",
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": -350,
                  },
                ],
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4292995566,
                  4294113532,
                  4292995566,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 20,
                "width": 650,
              },
              Object {
                "backgroundColor": "#E1E9EE",
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": -350,
                  },
                ],
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4292995566,
                  4294113532,
                  4292995566,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 25,
        }
      }
    >
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 50,
                "height": 30,
                "marginRight": 20,
                "width": 30,
              },
              Object {
                "backgroundColor": "#E1E9EE",
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": -350,
                  },
                ],
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4292995566,
                  4294113532,
                  4292995566,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 20,
                "width": 650,
              },
              Object {
                "backgroundColor": "#E1E9EE",
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": -350,
                  },
                ],
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4292995566,
                  4294113532,
                  4292995566,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
